sdstore/cache2disk.go

```diff
@@ -177,201 +252,199 @@ get cache from shared file server to local
 		destPath = filepath.Dir(dest)
 		srcPath = filepath.Dir(src)
 		srcFile = filepath.Base(src)
 	}
 
-	newMd5, sizeInBytes := getMetadataInfo(src)
+	fInfos, newMd5, sizeInBytes := getMetadataInfo(src)
 	if cacheMaxSizeInMB > 0 {
 		cacheMaxSizeInBytes := cacheMaxSizeInMB << (10 * 2) // MB to Bytes
 		fmt.Printf("size: %v B\n", sizeInBytes)
 		if sizeInBytes > cacheMaxSizeInBytes {
-			msg = fmt.Sprintf("source directory size %v B is more than allowed max limit %v B", sizeInBytes, cacheMaxSizeInBytes)
-			return logger.Log(logger.LoglevelError, "", logger.ErrtypeMaxsizelimit, msg)
+			return logger.Error(fmt.Errorf("source directory size %v B is more than allowed max limit %v B", sizeInBytes, cacheMaxSizeInBytes))
 		}
-		_ = logger.Log(logger.LoglevelInfo, "", fmt.Sprintf("source directory size %vB, allowed max limit %vB", sizeInBytes, cacheMaxSizeInBytes))
+		logger.Info(fmt.Sprintf("source directory size %vB, allowed max limit %vB", sizeInBytes, cacheMaxSizeInBytes))
 	}
 
-	_ = logger.Log(logger.LoglevelInfo, "", fmt.Sprintf("md5 check %v", md5Check))
-	if md5Check {
-		if compareMd5(newMd5, destPath, destBase) {
-			return logger.Log(logger.LoglevelWarn, "", logger.ErrtypeFile, fmt.Sprintf("source %s and destination %s directories are same, aborting", src, dest))
-		}
+	if compareMd5(newMd5, destPath, destBase) {
+		logger.Warn(fmt.Sprintf("source %s and destination %s directories are same, aborting", src, dest))
+		return nil
 	}
 
-	if compress {
-		targetPath := fmt.Sprintf("%s%s", filepath.Join(destPath, destBase), CompressFormatTarZst)
-		cwd, err = os.Getwd()
-		if err != nil {
-			return logger.Log(logger.LoglevelError, "", logger.ErrtypeZip, err)
-		}
-		_ = os.MkdirAll(destPath, 0777)
-		cmd := fmt.Sprintf("cd %s && tar -c %s | %s -T0 %s > %s || true; cd %s", srcPath, srcFile, getZstdBinary(), CompressionLevel, targetPath, cwd)
-		err = ExecuteCommand(cmd)
-		if err != nil {
-			msg = fmt.Sprintf("failed to compress files from %v", src)
-			_ = logger.Log(logger.LoglevelWarn, "", logger.ErrtypeZip, msg)
-		}
-		_ = os.Chmod(targetPath, 0777)
-		_ = os.Chmod(destPath, 0777)
-
-		// remove zip file if available
-		targetPath = fmt.Sprintf("%s%s", filepath.Join(destPath, destBase), CompressFormatZip)
-		defer os.RemoveAll(targetPath)
-	} else {
-		if err = copy.Copy(src, dest); err != nil {
-			return logger.Log(logger.LoglevelError, "", logger.ErrtypeCopy, err)
-		}
+	targetPath := fmt.Sprintf("%s%s", filepath.Join(destPath, destBase), CompressFormatTarZst)
+	cwd, err = os.Getwd()
+	if err != nil {
+		return logger.Error(err)
 	}
+	_ = os.MkdirAll(destPath, DefaultFilePermission)
 
-	if md5Check {
-		md5Path = filepath.Join(destPath, fmt.Sprintf("%s%s", destBase, Md5Extension))
-		md5File, err = os.OpenFile(md5Path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0777)
-		if err != nil {
-			_ = logger.Log(logger.LoglevelWarn, "", logger.ErrtypeMd5, fmt.Sprintf("not able to create %v file", md5Path))
+	if ZstdCli {
```

Comment:
can we assume is zstdcli is always available ? If not is it using just `zip` ?

Reply:
yes, zstdcli is available - https://github.com/screwdriver-cd/launcher/blob/b33f0bd0bb016a5fa1ab21b6a0b1294876097bf7/Dockerfile#L94

otherwise, it will use the zstd go library, next release will remove the zstd cli